signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/multikey/PublicKeyFileSelector.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.secp256k1.multikey;
+
+import tech.pegasys.signers.secp256k1.api.FileSelector;
+import tech.pegasys.signers.secp256k1.api.PublicKey;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream.Filter;
+import java.nio.file.Path;
+import java.util.List;
+
+import com.google.common.base.Splitter;
+
+/*
+Sample FileSelector which accepts:
+* files with an extension of toml (case insensitive)
+* first part of filename is the public keys hex string representation (without 0x prefix), case insensitive
+* Ignores interim "dotted" elements (eg <publicKey>.arbitrary.data.toml is valid)
+ */
+public class PublicKeyFileSelector implements FileSelector<PublicKey> {
+
+  final String fileExtension = "toml";
+
+  @Override
+  public Filter<Path> getAllConfigFilesFilter() {
+    return this::hasExpectedFileExtension;
+  }
+
+  @Override
+  public Filter<Path> getSpecificConfigFileFilter(final PublicKey selectionCriterion) {
+    return entry -> matchesPublicKey(selectionCriterion, entry);
+  }
+
+  public boolean matchesPublicKey(final PublicKey publicKey, final Path entry) throws IOException {
```

Comment:
Doesn't look like the throws for IOException is needed

Reply:
agreed - but the fileFilter run as part of the getSpecificConfiFileFilter _can_ throw an IOException as part of its inherited function ... how do we avoid that?